File: calculatorApp/calculatorApp/ViewController.m
//
//  ViewController.m
//  calculatorApp
//
//  Created by Merritt Tidwell on 1/20/15.
//  Copyright (c) 2015 Merritt Tidwell. All rights reserved.
//

#import "ViewController.h"

int Method;
float runningTotal;
int selectedNumber;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}



-(IBAction)btnPressed:(UIButton *)sender{

    
    NSLog(@"button was pressed - %@", sender.titleLabel.text);

    NSString * inputString = sender.titleLabel.text;
    selectedNumber = [inputString intValue];
   
    _numberLabel.text = [NSString stringWithFormat:@"%d", selectedNumber] ;
    
    
    
    

}

- (IBAction)divideButtonPressed:(id)sender {

    if (runningTotal == 0) {
        runningTotal = selectedNumber;
    }else{
        switch (Method) {
            case 1:
                runningTotal = runningTotal / selectedNumber;
                break;
            case 2:
                runningTotal = runningTotal * selectedNumber;
                break;
            case 3:
                runningTotal = runningTotal - selectedNumber;
                break;
            case 4:
                runningTotal = runningTotal + selectedNumber;
                break;
            default:
                break;
                
        }
    }
    
    
    
    Method = 1;
        selectedNumber = 0;


}

- (IBAction)timesButtonPressed:(id)sender {

    
    
    
    if (runningTotal == 0) {
        runningTotal = selectedNumber;
    }else{
        switch (Method) {
            case 1:
                runningTotal = runningTotal / selectedNumber;
                break;
            case 2:
                runningTotal = runningTotal * selectedNumber;
                break;
            case 3:
                runningTotal = runningTotal - selectedNumber;
                break;
            case 4:
                runningTotal = runningTotal + selectedNumber;
                break;
            default:
              break;
        
        }
        }
    
    Method = 2;
    selectedNumber = 0;





}

- (IBAction)subtractButtonPressed:(id)sender {

    
    if (runningTotal == 0) {
        runningTotal = selectedNumber;
    }else{
        switch (Method) {
            case 1:
                runningTotal = runningTotal / selectedNumber;
                break;
            case 2:
                runningTotal = runningTotal * selectedNumber;
                break;
            case 3:
                runningTotal = runningTotal - selectedNumber;
                break;
            case 4:
                runningTotal = runningTotal + selectedNumber;
                break;
            default:
                break;
                
        }
    }
    
    
    Method = 3;
    selectedNumber = 0;

}

- (IBAction)plusButtonPressed:(id)sender {

    if (runningTotal == 0) {
        runningTotal = selectedNumber;
    }else{
        switch (Method) {
            case 1:
                runningTotal = runningTotal / selectedNumber;
                break;
            case 2:
                runningTotal = runningTotal * selectedNumber;
                break;
            case 3:
                runningTotal = runningTotal - selectedNumber;
                break;
            case 4:
                runningTotal = runningTotal + selectedNumber;
                break;
            default:
                break;
                
        }

    }
    
    Method = 4;
    selectedNumber = 0;

}

- (IBAction)equalButtonPressed:(id)sender {


    if (runningTotal == 0) {
        runningTotal = selectedNumber;
    }else{
        switch (Method) {
            case 1:
                runningTotal = runningTotal / selectedNumber;
                break;
            case 2:
                runningTotal = runningTotal * selectedNumber;
                break;
            case 3:
                runningTotal = runningTotal - selectedNumber;
                break;
            case 4:
                runningTotal = runningTotal + selectedNumber;
                break;
            default:
                break;
                
        }

    
    }
    
    Method = 0;
    selectedNumber = 0;
    _numberLabel.text = [NSString stringWithFormat:@"%.1f", runningTotal];


}
- (IBAction)allClearPressed:(id)sender {

    Method = 0;
    runningTotal = 0;
    _numberLabel.text = [NSString stringWithFormat:@"0"];



}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: My Data/My Data/ViewController.swift
//
//  ViewController.swift
//  My Data
//
//  Created by Merritt Tidwell on 9/2/14.
//  Copyright (c) 2014 Merritt Tidwell. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
   
        
    
        var appD = UIApplication.sharedApplication().delegate as AppDelegate
        
        
        var entity = NSEntityDescription.entityForName("Post", inManagedObjectContext: appD.managedObjectContext)
        
        
        var postObject = NSEntityDescription.insertNewObjectForEntityForName("Post", inManagedObjectContext: appD.managedObjectContext) as NSManagedObject
        
        postObject.setValue("jo@theironyard.com", forKey: "user")
        postObject.setValue("this is a cool title", forKey: "title")
        
//       appD.saveContext()
        
        ///// fetch
        
        
        var fetchRequest = NSFetchRequest ()
       
        
        fetchRequest.entity = entity
        
        var predicate = NSPredicate(format: "user = 'jo@theironyard.com'" )
    
        fetchRequest.predicate = predicate
        
        var sortDescriptor = NSSortDescriptor(key: "posted_at", ascending: true)
        
        fetchRequest.sortDescriptors = [sortDescriptor]
        
        var fetchedObjects = appD.managedObjectContext!.executeFetchRequest(fetchRequest, error: nil)
        
        println(fetchedObjects)

        
        
    
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

